utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,726 @@
+    hipHostMalloc(&srcImgSizes, noOfImages * sizeof(RpptImagePatch));
+    hipHostMalloc(&dstImgSizes, noOfImages * sizeof(RpptImagePatch));
+
+    // Set ROI tensors types for src/dst
+    RpptRoiType roiTypeSrc, roiTypeDst;
+    roiTypeSrc = RpptRoiType::XYWH;
+    roiTypeDst = RpptRoiType::XYWH;
+
+    // Set maxHeight, maxWidth and ROIs for src/dst
+    const int images = noOfImages;
+
+    for(int i = 0; i < imageNames.size(); i++)
+    {
+        string temp = src1;
+        temp += imageNames[i];
+        if (layoutType == 0 || layoutType == 1)
+            image = imread(temp, 1);
+        else
+            image = imread(temp, 0);
+
+        roiTensorPtrSrc[i].xywhROI = {0, 0, image.cols, image.rows};
+        roiTensorPtrDst[i].xywhROI = {0, 0, image.cols, image.rows};
+
+        srcImgSizes[i].width = roiTensorPtrSrc[i].xywhROI.roiWidth;
+        srcImgSizes[i].height = roiTensorPtrSrc[i].xywhROI.roiHeight;
+        dstImgSizes[i].width = roiTensorPtrDst[i].xywhROI.roiWidth;
+        dstImgSizes[i].height = roiTensorPtrDst[i].xywhROI.roiHeight;
+
+        maxHeight = std::max(maxHeight, roiTensorPtrSrc[i].xywhROI.roiHeight);
+        maxWidth = std::max(maxWidth, roiTensorPtrSrc[i].xywhROI.roiWidth);
+        maxDstHeight = std::max(maxDstHeight, roiTensorPtrDst[i].xywhROI.roiHeight);
+        maxDstWidth = std::max(maxDstWidth, roiTensorPtrDst[i].xywhROI.roiWidth);
+
+        count++;
+    }
+
+    // Check if any of maxWidth and maxHeight is less than or equal to 0
+    if(maxHeight <= 0 || maxWidth <= 0)
+    {
+        std::cerr<<"Unable to read images properly.Please check the input path of the files specified";
+        exit(0);
+    }
+
+    Rpp32u outputChannels = inputChannels;
+    if(pln1OutTypeCase)
+        outputChannels = 1;
+    Rpp32u offsetInBytes = 0;
+
+    // Set numDims, offset, n/c/h/w values, strides for src/dst
+    set_descriptor_dims_and_strides(srcDescPtr, noOfImages, maxHeight, maxWidth, inputChannels, offsetInBytes);
+    set_descriptor_dims_and_strides(dstDescPtr, noOfImages, maxDstHeight, maxDstWidth, outputChannels, offsetInBytes);
+
+    // Set buffer sizes in pixels for src/dst
+    ioBufferSize = (Rpp64u)srcDescPtr->h * (Rpp64u)srcDescPtr->w * (Rpp64u)srcDescPtr->c * (Rpp64u)noOfImages;
+    oBufferSize = (Rpp64u)dstDescPtr->h * (Rpp64u)dstDescPtr->w * (Rpp64u)dstDescPtr->c * (Rpp64u)noOfImages;
+
+    // Set buffer sizes in bytes for src/dst (including offsets)
+    Rpp64u ioBufferSizeInBytes_u8 = ioBufferSize + srcDescPtr->offsetInBytes;
+    Rpp64u oBufferSizeInBytes_u8 = oBufferSize + dstDescPtr->offsetInBytes;
+    Rpp64u inputBufferSize = ioBufferSize * get_size_of_data_type(srcDescPtr->dataType) + srcDescPtr->offsetInBytes;
+    Rpp64u outputBufferSize = oBufferSize * get_size_of_data_type(dstDescPtr->dataType) + dstDescPtr->offsetInBytes;
+
+    // Initialize 8u host buffers for src/dst
+    Rpp8u *inputu8 = (Rpp8u *)calloc(ioBufferSizeInBytes_u8, 1);
```

Comment:
please use static_cast<> instead, same true for following. None of the code should contain c_type casts